libsolidity/interface/StandardCompiler.cpp

```diff
@@ -613,12 +620,20 @@ std::variant<OptimiserSettings, Json::Value> parseOptimizerSettings(Json::Value
 			return *error;
 		if (auto error = checkOptimizerDetail(details, "constantOptimizer", settings.runConstantOptimiser))
 			return *error;
-		if (auto error = checkOptimizerDetail(details, "yul", settings.runYulOptimiser))
+
+		if (auto error = checkOptimizerDetail(details, "yul", yulOptimizerRequested))
 			return *error;
-		settings.optimizeStackAllocation = settings.runYulOptimiser;
+
+		if (yulOptimizerRequested)
+			settings.copyYulSettingsFrom(OptimiserSettings::standard());
+		else
+			// NOTE: We want Yul optimizer to run regardless of the `yul` setting, just with minimal steps.
+			// This is the minimum necessary to avoid running into "Stack too deep" constantly.
+			settings.copyYulSettingsFrom(OptimiserSettings::minimal());
```

Comment:
Note that one of the effects of this change is that in metadata we'll always end up with `yul: true` even if the user sets it to `false` in settings. This should not affect reproducibility because we'll also have the optimizer sequence there.

As long as no one assumes that the sequence may not be present when `yul: false`, it should be fine.

Reply:
Marking as resolved. Now that I don't modify settings, we end with it properly marked in metadata (though on the other hand now the flag no longer means that we did not run Yul optimizer; but that's a different matter).